### PR TITLE
std.c: freebsd add procctl exclusive x86_64 flags

### PR DESCRIPTION
**Pull request by devnexen**

### CODE REVIEW

File: lib/std/c/freebsd.zig [hunk 1]
```diff
@@ -2172,6 +2172,56 @@ pub const PROC = struct {
     pub const WX_MAPPINGS_PERMIT = 0x0001;
     pub const WX_MAPPINGS_DISALLOW_EXEC = 0x0002;
     pub const WX_MAPPINGS_ENFORCE = 0x80000000;
+    pub const PROCCTL_MD_MIN = 0x10000000;
+    // specific x86_64 constants
```
Comment (matu3ba) [inline lib/std/c/freebsd.zig:2176]: s/specific x86_64 constants/x86_64-only constants

File: lib/std/c/freebsd.zig [hunk 2]
```diff
@@ -2172,6 +2172,56 @@ pub const PROC = struct {
...
+    pub const KPTI_CTL = switch (builtin.cpu.arch) {
```
Comment (matu3ba) [inline lib/std/c/freebsd.zig:2177]: That are alot of switch else branches cluttering the code. It looks better to me to namespace `KPTI` and `LA`:
```zig
pub const KPTI = switch (builtin.cpu.arch) {
    .x86_64 => struct {
        pub const CTL = PROCCTL_MD_MIN,
        // ..
    },
    else => void,
};
```

FreeBSD supports [many platforms](https://www.freebsd.org/platforms/). At which points do you think should we separate out the platform-specific bits?

Comment (devnexen) [reply lib/std/c/freebsd.zig:2177]: thanks for the tip, for few years there only had been those few x86 64 constants, nothing for arm or anything else (yet).

File: lib/std/c/freebsd.zig [hunk 3]
```diff
@@ -2172,6 +2172,30 @@ pub const PROC = struct {
...
+    };
+    pub const LA = switch (builtin.cpu.arch) {
+        .x86_64 => struct {
+            pub const CTL = PROC.PROCCTL_MD_MIND + 2;
+            pub const STATUS = PROC.PROCCTL_MD_MIND + 3;
+            pub const CTL_LA48_ON_EXEC = 1;
+            pub const CTL_LA57_ON_EXEC = 2;
+            pub const CTL_DEFAULT_ON_EXEC = 3;
+            pub const STATUS_LA48 = 0x01000000;
+            pub const STATUS_LA57 = 0x01000000;
```
Comment (matu3ba) [inline lib/std/c/freebsd.zig:2195]: STATUS_LA48 == STATUS_LA57 look abit odd. You could mention that this is, but you dont need to.

**Review by matu3ba — APPROVED** [review]

File: lib/std/c/freebsd.zig (hunk 3 above)
Comment (devnexen) [reply lib/std/c/freebsd.zig:2195]: not that s simply a typo.

**Comment by devnexen** [comment]
cc @Vexu 🙂

**Pull request merged**